### PR TITLE
Unregister the activity lifecycle callbacks when disposing the camera

### PR DESCRIPTION
**Pull request by kmorkos**
Fix crash referenced in https://github.com/flutter/flutter/issues/18348

### CODE REVIEW

**Comment by kmorkos** [comment]
I signed it.

**Comment by kmorkos** [comment]
I signed it

**Comment by mvwaran** [comment]
Is this fixed? I'm still facing the issue and i'm using [camera plugin 0.2.3 version](https://pub.dartlang.org/packages/camera).

**Comment by kmorkos** [comment]
It was never merged in... I'm not sure how to get a reviewer to look at it.

**Comment by kmorkos** [comment]
@kroikie I saw you were assigned to some other pull requests. Perhaps you can take a look at this or tag somebody else who can?

**Comment by bparrishMines** [comment]
Hi @kmorkos 

Everything looks good except I think you should re-register the `activitylifecyclecallbacks` in the 'initialize' method. I think an error would occur if one camera was disposed and then another was initialized without the `activitylifecyclecallback` being registered.

File: packages/camera/android/src/main/java/io/flutter/plugins/camera/CameraPlugin.java [hunk 1]
```diff
@@ -76,53 +77,53 @@ private CameraPlugin(Registrar registrar, FlutterView view, Activity activity) {
...
+          public void onActivitySaveInstanceState(Activity activity, Bundle outState) {}
 
-              @Override
-              public void onActivityDestroyed(Activity activity) {}
-            });
+          @Override
+          public void onActivityDestroyed(Activity activity) {}
+        };
+
+    activity.getApplication().registerActivityLifecycleCallbacks(this.activityLifecycleCallbacks);
```
Comment (bparrishMines) [inline packages/camera/android/src/main/java/io/flutter/plugins/camera/CameraPlugin.java]: Could this be moved to the `initialized` method call?

**Comment by kmorkos** [comment]
@bparrishMines thanks for checking this out. That makes sense, just made that change!

**Comment by kmorkos** [comment]
@bparrishMines thanks! Can you push the update to pub please? :smile:

**Comment by bparrishMines** [comment]
@kmorkos Sorry about that! Published!

**Pull request merged**